### DISCUSSION TITLE
Use markdown sources

### DISCUSSION THREAD

**Discussion by memowe** [learnyouahaskell/learnyouahaskell.github.io#10] [Ideas]
*I wrote this as issue #8 before but maybe it's better to go one step back and use this platform to discuss a strategy before.*

> Thanks for all your effort in this important project!
> 
> I think it would be great to have the book‘s content as markdown files instead of HTML files:
> 
> **Pros**:
> 
> 1. It would make contributions much easier as it wouldn‘t require web authoring skills.
> 2. It would make diffs much more readable.
> 3. With pandoc and custom templates it would make exporting HTML, PDF or EPUB a very transparent process.
> 
> **Cons**:
> 
> 1. Lots of work. Especially when it‘s a requirement to produce (almost) the same HTML result as before.
> 2. ...?
> 
> What do you think about it?

I'm willing to help if the community thinks this would be the right thing to do. Im looking forward to ideas and opinions. :)

**Comment by simon-nicholls** [comment]
I've only just become aware of this community effort to update LYAH, but am stoked it's a thing. Huge thanks to Miran Lipovača!

My thought is to focus on facilitating easy community participation first, and move to Markdown. Do we have the right to produce updated ebooks?

**Comment by ShaharNaveh** [comment]
I do think that it would be easier to maintain markdown instead of raw HTML.
I recommend the approach of using a static-site-generator like Hugo or Jekyll for example.
It has the benefit of using existing themes, a lot of them support features that we currently do no have such as:
- toggling dark/light mode
- copy button for codeblocks
- multi lingual support (if there's a need for that)
- mobile support

And probably much more.

---

I do not have experience exporting static sites to PDF/epub, but from searching it online it seems relatively straightforward by creating a custom page template that contains all the site's pages and then converting the generated HTML to the desired format(s) with a tool like pandoc

**Comment by ulysses4ever** [comment]
This has been implemented #103.

**Comment by AvverbioPronome** [comment]
You said epub?

I started something but it's not release-ready: https://github.com/AvverbioPronome/learnyouahaskell.github.io/tree/epub

**Reply by ulysses4ever** [reply]
Exciting! Please submit a PR whenever ready!
